File: backend_py/google_docs.py
import json
from google.oauth2.credentials import Credentials
from googleapiclient.discovery import build
from config import GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET
from google_calendar import SCOPES

def create_google_doc(credentials_json: str, title: str, content: str) -> str:
    cred = json.loads(credentials_json)
    refresh_token = cred.get("refresh_token")
    if not refresh_token:
        raise ValueError("Missing refresh_token")
    creds = Credentials(token=None, refresh_token=refresh_token, token_uri="https://oauth2.googleapis.com/token", client_id=GOOGLE_CLIENT_ID, client_secret=GOOGLE_CLIENT_SECRET, scopes=SCOPES)
    docs = build("docs", "v1", credentials=creds)
    create_res = docs.documents().create(body={"title": title or "Untitled"}).execute()
    document_id = create_res.get("documentId")
    if not document_id:
        raise ValueError("Failed to create document")
    if content and content.strip():
        docs.documents().batchUpdate(documentId=document_id, body={
            "requests": [{"insertText": {"location": {"index": 1}, "text": content.strip()}}]
        }).execute()
    return f'Document created: "{title}". View: https://docs.google.com/document/d/{document_id}/edit'

File: backend/src/google-docs.ts
import { google } from 'googleapis';
import { config } from './config.js';

function getOAuth2Client() {
  if (!config.googleClientId || !config.googleClientSecret) {
    throw new Error('GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET must be set for Google');
  }
  const redirectUri = config.googleRedirectUri || `http://localhost:${config.port}/auth/google/callback`;
  return new google.auth.OAuth2(config.googleClientId, config.googleClientSecret, redirectUri);
}

export async function createGoogleDoc(
  credentialsJson: string,
  title: string,
  content: string
): Promise<string> {
  let cred: { refresh_token: string };
  try {
    cred = JSON.parse(credentialsJson) as { refresh_token: string };
  } catch {
    throw new Error('Invalid Google credentials');
  }
  if (!cred.refresh_token) throw new Error('Missing refresh_token');
  const oauth2 = getOAuth2Client();
  oauth2.setCredentials({ refresh_token: cred.refresh_token });
  const docs = google.docs({ version: 'v1', auth: oauth2 });

  const createRes = await docs.documents.create({
    requestBody: { title: title || 'Untitled' },
  });
  const documentId = createRes.data.documentId;
  if (!documentId) throw new Error('Failed to create document');

  if (content && content.trim()) {
    await docs.documents.batchUpdate({
      documentId,
      requestBody: {
        requests: [
          {
            insertText: {
              location: { index: 1 },
              text: content.trim(),
            },
          },
        ],
      },
    });
  }

  const link = `https://docs.google.com/document/d/${documentId}/edit`;
  return `Document created: "${title}". View: ${link}`;
}
